### PR TITLE
Add appr client

### PR DESCRIPTION
**Pull request by fgimenez**
Towards https://github.com/giantswarm/giantswarm/issues/1902

Initial structure and configuration for the appr client, also wired with the chart resource.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -137,6 +137,7 @@ func mainWithError() (err error) {
 	daemonCommand := newCommand.DaemonCommand().CobraCommand()
 
 	daemonCommand.PersistentFlags().String(f.Service.Kubernetes.Address, "", "Address used to connect to Kubernetes. When empty in-cluster config is created.")
+	daemonCommand.PersistentFlags().String(f.Service.CNR.Address, "https://quay.io", "Address used to connect to CNR, defaults to quay's managed offering.")
```
Comment (rossf7) [inline main.go]: Makes sense since we use Quay and its the only managed service at the moment.

File: service/chartconfig/chart_framework.go [hunk 2]
```diff
@@ -35,6 +37,9 @@ func NewChartFramework(config ChartFrameworkConfig) (*framework.Framework, error
 	if config.K8sExtClient == nil {
 		return nil, microerror.Maskf(invalidConfigError, "config.K8sExtClient must not be empty")
 	}
+	if config.ApprClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.ApprClient must not be empty")
+	}
```
Comment (xh3b4sd) [inline service/chartconfig/chart_framework.go]: IMO we should only validate where we actually use something. The app client is going to be validated at the end of the chain anyway. I think it is not necessary to validate in each layer. This is just unnecessary code.

File: service/chartconfig/chart_framework.go [hunk 3]
```diff
@@ -12,12 +12,14 @@ import (
...
 
 	"github.com/giantswarm/chart-operator/service/chartconfig/v1"
+	"github.com/giantswarm/chart-operator/service/chartconfig/v1/appr"
 )
 
 type ChartFrameworkConfig struct {
 	G8sClient    versioned.Interface
 	K8sClient    kubernetes.Interface
 	K8sExtClient apiextensionsclient.Interface
+	ApprClient   *appr.Client
```
Comment (xh3b4sd) [inline service/chartconfig/chart_framework.go]: A comes usually before K and G.

File: service/chartconfig/v1/appr/appr.go [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+// Config represents the configuration used to create a appr client.
+type Config struct {
+	Logger  micrologger.Logger
+	Address string
+}
+
+// Client knows how to talk with a CNR server.
+type Client struct {
+	logger  micrologger.Logger
+	address string
```
Comment (xh3b4sd) [inline service/chartconfig/v1/appr/appr.go]: Dependencies and setting should have separate blocks.

**Comment by fgimenez** [comment]
Thanks @xh3b4sd I'll address your comments in a followup.

**Pull request merged**